lib/jwtutil/jwtutil.go

```diff
@@ -0,0 +1,90 @@
+package jwtutil
+
+import (
+	"bytes"
+	"crypto/rand"
+	"crypto/rsa"
+	"encoding/hex"
+	"encoding/json"
+	"fmt"
+	"log"
+	"net/http"
+	"time"
+
+	"github.com/rclone/rclone/fs/config/configmap"
+	"github.com/rclone/rclone/lib/oauthutil"
+
+	"golang.org/x/oauth2"
+	"golang.org/x/oauth2/jws"
+)
+
+// RandomHex creates a random string of the given length
+func RandomHex(n int) (string, error) {
+	bytes := make([]byte, n)
+	if _, err := rand.Read(bytes); err != nil {
+		return "", err
+	}
+	return hex.EncodeToString(bytes), nil
+}
+
+// Config configures rclone using JWT
+func Config(id, name string, claims *jws.ClaimSet, header *jws.Header, queryParams map[string]string, privateKey *rsa.PrivateKey, m configmap.Mapper) error {
+	payload, err := jws.Encode(header, claims, privateKey)
+	if err != nil {
+		log.Fatalf("Failed to encode payload: %v", err)
+	}
+	req, err := http.NewRequest("POST", claims.Aud, nil)
+	if err != nil {
+		log.Fatalf("Failed to create new request: %v", err)
+	}
+	q := req.URL.Query()
+	q.Add("grant_type", "urn:ietf:params:oauth:grant-type:jwt-bearer")
+	q.Add("assertion", payload)
+	for key, value := range queryParams {
+		q.Add(key, value)
+	}
+	queryString := q.Encode()
+
+	req, err = http.NewRequest("POST", claims.Aud, bytes.NewBuffer([]byte(queryString)))
+	if err != nil {
+		log.Fatalf("Failed to create new request: %v", err)
+	}
+	req.Header.Set("Content-Type", "application/x-www-form-urlencoded")
+
+	client := &http.Client{}
```

Comment:
We should probably pass in a *http.Client here so the caller can pass in [an rclone client](https://github.com/rclone/rclone/blob/master/fs/fshttp/http.go#L188) which will do debuging, rate limiting etc, etc